apps/api/src/helpers/lnurl.js

```diff
@@ -99,3 +102,78 @@ export const createLnurlAuthPubkeyAndSignature = async ({
     return Promise.reject(error)
   }
 }
+
+export const createLnurlAuthPubkeyAndSignature = async ({
+  lnurl,
+  withJWT = false,
+}) => {
+  try {
+    if (!lnurl) {
+      throw new Error('lnurlNotFound')
+    }
+
+    const { words } = bech32.decode(lnurl, 500)
+    const bytes = bech32.fromWords(words)
+    const string = Buffer.from(bytes).toString()
+
+    const url = new URL(string)
+    const { k1, hmac } = Object.fromEntries(url.searchParams)
+
+    if (!url.host || !k1) {
+      throw new Error('MissingInformations')
+    }
+
+    const { publicKey, privateKey } = await generateKeys(url.host)
+
+    const sign = secp256k1.ecdsaSign(
+      Buffer.from(k1, 'hex'),
+      Buffer.from(privateKey, 'hex')
+    )
+
+    const signature = secp256k1.signatureExport(sign.signature)
+    const hexStringSignature = arrayToHexString(signature)
```

Comment:
Why not simply use `Buffer.from(signature).toString('hex')` like in the backend?

Reply:
It's outdated code that is being replaced by the new pubkey / signature function.